File: DSA/Sliding window/subarray.js
function isPairSum(A, N, X)
{
 
    // represents first pointer
    var i = 0;
 
    // represents second pointer
    var j = N - 1;
 
    while (i < j) {
 
        // If we find a pair
        if (A[i] + A[j] == X)
            return true;
 
        // If sum of elements at current
        // pointers is less, we move towards
        // higher values by doing i++
        else if (A[i] + A[j] < X)
            i++;
 
        // If sum of elements at current
        // pointers is more, we move towards
        // lower values by doing j--
        else
            j--;
    }
    return false;
}
 
// Driver code
 
    // array declaration
    var arr = [ 2, 3, 5, 8, 9, 10, 11 ];
     
    // value to search
    var val = 17;
     
    // size of the array
    var arrSize =7;
     console.log(isPairSum(A, N, X));

File: MyDsa/Array/subArray.java
package Array;

public class subArray {

    public static void main(String[] args) {

        int[] arr = { 1, 2, 3 };
        int len = arr.length;

        for (int i = 0; i < len; i++) {
            int[] ar = new int[i + 1];
            for (int j = 0; j <= i; j++) {
                ar[j] = arr[j];
            }
            System.out.println(ar);
        }
    }
}
